client/src/includes/a11y-result.ts

```diff
@@ -70,6 +72,83 @@ export const getAxeConfiguration = (
   return null;
 };
 
+/**
+ * Custom rule for checking image alt text. This rule checks if the alt text for images
+ * contains poor quality text like file extensions or URLs.
+ * The rule will be added via the Axe.configure() API.
+ * https://github.com/dequelabs/axe-core/blob/master/doc/API.md#api-name-axeconfigure
+ */
+export const checkImageAltText = (node: HTMLImageElement, options) => {
+  if (!options.pattern) return undefined;
+
+  const altTextAntipatterns = new RegExp(options.pattern, 'i');
+  const altText = node.getAttribute('alt') || '';
+
+  const hasBadAltText = altTextAntipatterns.test(altText);
+  return !hasBadAltText;
+};
+
+/**
+ * Defines custom Axe rules, mapping each check to its corresponding JavaScript function.
+ * This object holds the custom checks that will be added to the Axe configuration.
+ */
+export const customChecks = {
+  'check-image-alt-text': checkImageAltText,
+  // Add other custom checks here
+};
+
+/**
+ * Configures custom Axe rules by integrating the custom checks with their corresponding
+ * JavaScript functions. It modifies the provided configuration to include these checks.
+ */
+export const addCustomChecks = (spec: Spec): Spec => {
+  const modifiedChecks = spec?.checks?.map((check) => {
+    if (customChecks[check.id]) {
+      return {
+        ...check,
+        evaluate: customChecks[check.id],
+      };
+    }
+    return check;
+  });
+  return {
+    ...spec,
+    checks: modifiedChecks,
+  };
+};
+
+interface A11yReport {
+  results: AxeResults;
+  a11yErrorsNumber: number;
+}
+
+/**
+ * Get accessibility testing results from Axe based on the configurable custom spec, context, and options.
+ * It integrates custom rules into the Axe configuration before running the tests.
+ */
+export const getA11yReport = async (
+  config: WagtailAxeConfiguration,
+): Promise<A11yReport> => {
+  let spec = config.spec;
+  // Apply custom configuration for Axe. Custom 'check-image-alt-text' is enabled by default
+  if (spec) {
+    if (spec.checks) {
+      spec = addCustomChecks(spec);
+    }
+    axe.configure(spec);
+  }
```

Comment:
Based on changes to the Python code, it’d be great if we could change this to:

```suggestion
  axe.configure(addCustomChecks(spec));
```

Only if this does work with `{}` as a spec though (or `{ checks: [], rules: [] }`).